src/Tables.jl

```diff
@@ -1,92 +1,396 @@
 module Tables
 
 using Unidecode
-using GeoParams: AbstractMaterialParam, param_info
+using GeoParams: AbstractMaterialParam, param_info, LinearViscous, PowerlawViscous, DislocationCreep, DiffusionCreep, CompositeRheology, Parallel
 using ..Units
 using ..MaterialParameters: MaterialParamsInfo
 
-export Phase2Dict, Dict2LatexTable
+export detachFloatfromExponent, Phase2Dict, Dict2LatexTable, Phase2DictMd, Dict2MarkdownTable, ParameterTable
 
-# was tested with:
-# TestPh = (SetMaterialParams(Name="Viscous Matrix", Phase=1, Density=ConstantDensity(),CreepLaws = SetDislocationCreep("Quartz Diorite | Hansen & Carter (1982)")),
-#           SetMaterialParams(Name="Viscous Sinker", Phase=2, Density= PT_Density(),CreepLaws = LinearViscous(η=1e21Pa*s)),
-#           SetMaterialParams(Name="Viscous Bottom", Phase=3, Density= PT_Density(),CreepLaws = SetDislocationCreep("Diabase | Caristan (1982)")));
 
-"""
-Phase2Dict() puts all parameters of a phase in a dict.
-Dict2LatexTable() writes .tex file with all parameters from Phase2Dict() output in a table.
+""" 
+detachFloatfromExponent() returns the number of float decimals after the comma as Integer, the float number without the exponent as string 
+and the exponent after the "e" as string.
+The argument output returns "1" for "ex" if the input number has no exponent.
 
 """
+function detachFloatfromExponent(str::String)
+    s = lowercase(str)
+    if 'e' in s
+        s, ex = split(s, "e")
+    else
+        ex = "1"
+    end
 
+    dig = something(findfirst('.', reverse(s)), 1) - 1
+    
+    return dig, s, ex
+
+end
+
+
+"""
+Phase2Dict() puts all parameters of a phase s in a dict.
+"""
 function Phase2Dict(s)
-    # Dict has Key with Fieldname and Value with Tuple(value, symbol, unit)
-    fds = Dict{String,Tuple{String,String,String,String}}()
+    # Dict has Key with Fieldname and Value with Tuple(value, symbol, flowlaw keyword, number of phases, (Disl+Diff) or Lin, Disl or Diff)
+    fds = Dict{String,Tuple{String,String,String,String,String,String}}()
     refs = Dict{String,Tuple{String,String,String}}()
-    # Descriptions for every parameter that could occur in the table and their corresponding variable name(s) that is used in GeoParams
+    if !(typeof(s) <: Tuple)
+        s = (s,)
+    end
     phasecount = length(s)
     k = 1
     flowlawcount = 0
     # Checks all Phases 
     for i in 1:phasecount
         fieldnames = propertynames(s[i])
-        # Goes through all fields of a phase that are not "Name"
+        global Diff = 0
```

Comment:
Maybe I'm missing it, but do `Diff`, `Disl` and `Lin` need to be `global`? If there's not a specific reason, avoid using global variables

Reply:
I tried it without the global. It only worked when these variables were global. I also wondered why they need to be global but I couldn't figure it out.

Reply:
Making them local also doesn't work: "UndefVarError: Disl not defined".

Reply:
That would defeat the purpose of those variables since they count the occurences of DislCreep, DiffCreep and LinVisc per phase.
You marked the problem as resolved. Is that correct or just a misclick?

Reply:
I think for now we can leave it as it is since it's working.